src/huntsman/pocs/observatory.py

```diff
@@ -501,8 +501,8 @@ def take_dark_observation(self, bias=False, **kwargs):
         ObsClass = BiasObservation if bias else DarkObservation
         observation = ObsClass(position=position)
 
-        # Dark observations don't care if it's dark or not
-        safety_kwargs = {"ignore": ["is_dark"]}
+        # Dark observations don't care if it's dark or not, or if bad weather
```

Comment:
```suggestion
        # Dark observations don't care if it's dark or not, or if bad weather
        
        if not self.dome.is_closed:
                      self.dome.close()
```

Reply:
Hmmm good point. Or to be flexible, we could just add the `is_good_weather` to the list of ones to ignore if the dome is already closed. This would allow someone (for whatever reason) to take darks with the dome open.

Reply:
Thinking this is good enough for now:
```
        # Dark observations don't care if it's dark or not
        safety_kwargs = {"ignore": ["is_dark"]}

        # Can ignore weather safety if dome is closed
        with suppress(AttributeError):
            if self.dome.is_closed:
                safety_kwargs["ignore"].append("good_weather")
```
If there is no dome attribute then we *should* be in a fail safe mode and check for weather safety. This *should* work fine if using a simulated dome, but if not then it is a bug with the dome simulator.